org.eclipse.jdt.core/codeassist/org/eclipse/jdt/core/ICompletionEngine.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.jdt.core;
+
+import org.eclipse.jdt.internal.compiler.env.ICompilationUnit;
```

Comment:
The interfaces should remain in the `internal.codeassist` package, we don't need them to be so strongly supported APIs at the moment, being internal interfaces is enough (especially as javac is a fragment and can access them easily)